crates/utils/src/utils/markdown/spoiler_rule.rs

```diff
@@ -0,0 +1,109 @@
+// Custom Markdown plugin to manage spoilers.
+//
+// Matches the capability described in Lemmy UI:
+// https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/utils.ts#L159
+// that is based off of:
+// https://github.com/markdown-it/markdown-it-container/tree/master#example
+//
+// FORMAT:
+// Input Markdown: ::: spoiler THE_LEAD_IN\nTHE_HIDDEN_SPOILER\n:::\n
+// Output HTML: <details><summary>THE_LEAD_IN</summary><p>THE_HIDDEN_SPOILER</p></details>
+
+use markdown_it::{
+  parser::{
+    block::{BlockRule, BlockState},
+    inline::InlineRoot,
+  },
+  MarkdownIt,
+  Node,
+  NodeValue,
+  Renderer,
+};
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+#[derive(Debug)]
+struct SpoilerBlock {
+  call_to_action: String,
+}
```

Comment:
Can this spoiler functionality be upstreamed into markdown-it? I have no experience with parsers and would rather not maintain this.

Reply:
I do not think so, since:
* concept of a spoiler is custom functionality maintained by Lemmy UI: https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/utils.ts#L159.
* The JS `markdown-it-container` library does have this exact code in their README, but as an example of a plugin and the maintainers do not appear to have plans to support it as a "standard" plugin (see https://github.com/markdown-it/markdown-it/issues/552 created by one of the maintainers of Lemmy, which is so weird and coincidental!).

If you're still interested in having this extended plugin support, I can add more comments here about what is happening and why, with links back to `markdown-it.rs` library for any contributors to have an easier go at making changes. I am also not a parser expert as well (everything I learned was through making this PR) so I can most likely break this down for folks to understand.

Otherwise if you would, after seeing what is required, rather not support spoilers, emojis, or anything that would require a deviation from the GitHub markdown spec, that is also an option!

As an aside, I am hoping to contribute to Lemmy on a more regular basis so I also do not plan to disappear after contributing this, and can be around to help maintain; and I didn't pick this issue because I have a vested interest in markdown processing, but because it was listed as a good first issue 😄 I do not know if that helps with your decision on maintenance, but I thought I'd put that out there.

Sorry for the essay - please let me know what you think! If you'd rather the issue be closed at this time due to the burden of maintenance I would rather not polish this particular PR and further and move on to another issue that you think would be beneficial for me to tackle for Lemmy.